### DISCUSSION TITLE
Where is "null value" check box parameter  in debug mode ??? 4.14

### DISCUSSION THREAD

**Discussion by mhalvarez** [eclipse-birt/birt#1762] [HowTo]
[Parameter null value check box in debug mode.pdf](https://github.com/user-attachments/files/15817980/Parameter.null.value.check.box.in.debug.mode.pdf)

**Comment by speckyspooky** [comment]
Which parameter have you set to activate the debug mode?
I have never seen before this mode.

**Comment by mhalvarez** [comment]
Excuse my English, thank you very much for your quick interest.

It's "eclipse" debug mode, but I just need to know : 

How can i set to null  the second parameter here : ? when the first parameter is a date parameter type in birt report .

void org.eclipse.birt.report.engine.api.IEngineTask.setParameterValue(String arg0, Object arg1)


Thank you in advance ! Thomas

**Comment by speckyspooky** [comment]
I have tried to understand your request but it isn't so easy
because it seems that you are not talking directly about the report himself.

I understand that you try to set the parametervalue via integrated java-call.
At this situation if I understand correctly you will have 2 parameters and you will set the 2nd to "null" if the 1st is a "Date"-parameter.

So you have to implements to steps:

1. step: evaluate the 1st parameter of your report
- option 1: if you know the report where you need this logic you can implements it based on the report name
- option 2: verification of parameters with "IParameterAdapter.getParameterDefinitionTask()" - but I have never used this before

2. step: set the value to your report-parameter
`<your interface implementation>.setParameterValue("parameterName", null);
`

My be this will help you a little bit for your next steps.

**Reply by mhalvarez** [reply]
Hi Thomas, thank you for your reply and your time.

Yes, I'm working with "parametervalue via integrated java-call."
That's all ok, I made some changes in the code because I was having a certainly erratic behavior with "java.util.date" and some dates that I am not able to explain.

As I told you I made some changes to "my code" like using "java.sql.date" and now it looks all ok.

Thanks a lot!! 

![image](https://github.com/eclipse-birt/birt/assets/35996183/bbd8df00-9587-4e33-b4b6-6b8ffe272d85)

**Reply by speckyspooky** [reply]
Sounds very good - and I like your chart, it looks really good!

Small hint to java.util.date - at our BIRT project we use (in most cases, not over all) ibm.icu.util.date
because there are more supported options for special requirements.
